flask_appbuilder/security/manager.py

```diff
@@ -934,15 +934,15 @@ def _ldap_calculate_user_roles(
             # lookup registration role in flask db
             fab_role = self.find_role(registration_role_name)
             if fab_role:
-                user_role_objects.append(fab_role)
+                user_role_objects.add(fab_role)
```

Comment:
@fredthomsen what happens if the `AUTH_USER_REGISTRATION_ROLE` overlaps with one of the mapped roles?

Does the set() correctly de-dupe the role model objects which have the same name?

Reply:
Yes it does.  Since `fab_role` is just the role name the set de-dups them.  We can make this a part of the test case very easily.  I've updated the test case to add the test user to the `AUTH_USER_REGISTRATION_ROLE`.